flaml/model.py

```diff
@@ -1754,6 +1754,141 @@ def fit(self, X_train, y_train, budget=None, **kwargs):
+                "low_cost_init_value": False
+            },
+            "lags": {
+                "domain": tune.quniform(lower=1, upper=data_size[0]-(pred_horizon+1), q=1),
+                "init_value": 1
+            },
+        })
+        return space
+
+
+    def __init__(self, task=TS_FORECAST, **params):
+        super().__init__(task, **params)
+        self.hcrystaball_model = None
+
+    def transform_X(self, X):
+        X = X.copy()
+        cols = list(X)
+        if len(cols) == 1:
+            ds_col = cols[0]
+            X = pd.DataFrame(index=X[ds_col])
+        elif len(cols) > 1:
+            ds_col = cols[0]
+            exog_cols = cols[1:]
+            X = X[exog_cols].set_index(X[ds_col])
+        return X
+
+    def _fit(self, flaml_estimator, X_train, y_train, budget=None, **kwargs):
+        from hcrystalball.wrappers import get_sklearn_wrapper
+
+        X_train = self.transform_X(X_train)
+        X_train = self._preprocess(X_train)
+        params = self.params.copy()
+        lags = params.pop("lags")
+        optimize_for_horizon = params.pop("optimize_for_horizon")
+        estimator = flaml_estimator(task="regression", **params)
+        self.hcrystaball_model = get_sklearn_wrapper(estimator.estimator_class)
+        self.hcrystaball_model.lags = int(lags)
+        self.hcrystaball_model.fit(X_train, y_train)
+        if optimize_for_horizon == True:
+            model_li = []
+            for i in range(1, kwargs["period"]+1):
+                X_fit, y_fit = self.hcrystaball_model._transform_data_to_tsmodel_input_format(X_train, y_train, i)
+                self.hcrystaball_model.model.set_params(**estimator.params)
+                model = self.hcrystaball_model.model.fit(X_fit, y_fit)
+                model_li.append(model)
+            self._model = model_li
+        else:
+            X_fit, y_fit = self.hcrystaball_model._transform_data_to_tsmodel_input_format(X_train, y_train, kwargs["period"])
+            self.hcrystaball_model.model.set_params(**estimator.params)
+            model = self.hcrystaball_model.model.fit(X_fit, y_fit)
+            self._model = model
+
+
+    def predict(self, X_test):
+        if self._model is not None:
+            X_test = self._preprocess(X_test)
+            X_test = self.transform_X(X_test)
+            if isinstance(self._model, list):
+                assert (
+                    len(self._model) == len(X_test)
+                ), "Model is optimized for horizon, length of X_test must be equal to `period`."
+                preds = []
+                for i in range(1, len(self._model) + 1):
+                    X_pred, _ = self.hcrystaball_model._transform_data_to_tsmodel_input_format(X_test.iloc[:i, :])
+                    preds.append(self._model[i - 1].predict(X_pred)[-1])
+                forecast = pd.DataFrame(data=np.asarray(preds).reshape(-1, 1), columns=[self.hcrystaball_model.name], index=X_test.index)

```

Comment:
lint

Reply:
the Lint test says `.\flaml\model.py:1828:47: E226 missing whitespace around arithmetic operator` but I can't see where it occurs

Reply:
: